File: src/main/java/com/company/service/StudentService.java
package com.company.service;

import com.company.entities.Student;
import com.company.entities.Subject;
import com.company.repository.MarkForLessonRepository;
import com.company.repository.StudentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

@Service
public class StudentService {

    @Autowired
    private StudentRepository repository;
    @Autowired
    private MarkForLessonRepository markForLessonRepository;

    public List<Student> findAll(){
        List<Student> students = (ArrayList<Student>)repository.findAll();
        return students.stream().sorted(Comparator.comparing(Student::getId)).collect(Collectors.toList());
    }

    public void addStudent(Student student){
        repository.save(student);
    }

    public Student findStudentById(long id){
        Student student = repository.findById(id).get();
        return student;
    }

    public void removeStudent(long id){
        markForLessonRepository.deleteByStudent(findStudentById(id));
        repository.deleteById(id);
    }

    public void updateStudent(Student student){
        repository.save(student);
    }

    public void addSubjectToStudent(Student student, Subject subject){
        student.addSubject(subject);
        repository.save(student);
    }

    public void removeSubjectFromStudent(Student student, Subject subject){
        student.removeSubject(subject);
        repository.save(student);
    }
}

File: src/main/resources/static/app/service/student.service.js
angular
    .module("SchoolApp")
    .factory("StudentService", ["$http", function ($http) {
        var ALL_STUDENTS = "http://localhost:8080/student/findAll";
        var ADD_STUDENT = "http://localhost:8080/student/add";
        var UPDATE_STUDENT = "http://localhost:8080/student/update";
        var DELETE_STUDENT = "http://localhost:8080/student/remove/";

        var service = {};


        service.getAllStudents = function getAllStudents() {
            return $http.get(ALL_STUDENTS);
        };

        service.addStudent = function addStudent(student) {
            return $http.post(ADD_STUDENT, student)
        };

        service.updateStudent = function updateStudent(student) {
            return $http.post(UPDATE_STUDENT, student)
        };

        service.deleteStudent = function deleteStudent(student) {
            return $http.delete(DELETE_STUDENT + student.id)
        };

        return service;
    }]);